Code/Editor/TrackViewNewSequenceDialog.cpp

```diff
@@ -34,22 +35,107 @@ namespace
+        {
+            constexpr int MaxInputLength = 160;
+
+            if (input.isEmpty())
+            {
+                SetEnabled(false);
+                return QValidator::Acceptable;
+            }
+
+            bool isValid = false;
+            SetEnabled(true);
+            if (input.contains('/'))
+            {
+                SetToolTip("A sequence name cannot contain a '/' character");
+            }
+            if (input.length() > MaxInputLength)
+            {
+                SetToolTip(QString("A sequence name cannot exceed %1 characters").arg(MaxInputLength).toStdString().c_str());
+            }
+            else if (input == LIGHT_ANIMATION_SET_NAME)
+            {
+                SetToolTip("The sequence name " LIGHT_ANIMATION_SET_NAME " is reserved.\nPlease choose a different name");
+            }
+            else
+            {
+                bool isNewName = true;
+                for (unsigned int k = 0; k < GetIEditor()->GetSequenceManager()->GetCount(); ++k)
+                {
+                    CTrackViewSequence* pSequence = GetIEditor()->GetSequenceManager()->GetSequenceByIndex(k);
+                    const QString fullname = QString::fromUtf8(pSequence->GetName().c_str());
+
+                    if (fullname.compare(input, Qt::CaseInsensitive) == 0)
+                    {
+                        isNewName = false;
+                        break;
+                    }
+                }
+
+                if (!isNewName)
+                {
+                    SetToolTip("Sequence with this name already exists");
+                }
+                else
+                {
+                    isValid = true;
+                }
+            }
+
+            if (isValid)
+            {
+                SetToolTip("");
+                return QValidator::Acceptable;
+            }
+
+            return QValidator::Invalid;
+        }
+
+    private:
+
+        void SetEnabled(bool enable) const
+        {
+            m_parentDialog->ui->buttonBox->button(QDialogButtonBox::Ok)->setEnabled(enable);
+        }
+
+        void SetToolTip(const char* toolTipText) const
+        {
+            m_parentDialog->ui->NAME->setToolTip(toolTipText);
+        }
+
+        const CTVNewSequenceDialog* m_parentDialog;
+};
+
 // TrackViewNewSequenceDialog dialog
 CTVNewSequenceDialog::CTVNewSequenceDialog(QWidget* parent)
     : QDialog(parent)
     , ui(new Ui::CTVNewSequenceDialog)
     , m_inputFocusSet(false)
+    , m_validator(new CTVNewSequenceDialogValidator(this))
 {
     ui->setupUi(this);
     connect(ui->buttonBox, &QDialogButtonBox::accepted, this, &CTVNewSequenceDialog::OnOK);
     connect(ui->NAME, &QLineEdit::returnPressed, this, &CTVNewSequenceDialog::OnOK);
+    ui->NAME->setValidator(m_validator);
     setWindowTitle("Add New Sequence");
 
     OnInitDialog();
 }
 
 CTVNewSequenceDialog::~CTVNewSequenceDialog()
 {
+    delete m_validator;
```

Comment:
Not needed as QT take ownership of the ptr as you give CTVNewSequenceDialog as parent